system-configuration/src/network_reachability.rs

```diff
@@ -344,30 +344,49 @@ impl<T: Fn(ReachabilityFlags) + Sync + Send> NetworkReachabilityCallbackContext<
 /// libc::sockaddr_in6, depending on the passed in standard library SocketAddr.
 fn to_c_sockaddr(addr: SocketAddr) -> Box<libc::sockaddr> {
     let ptr = match addr {
+        // See reference conversion from socket2: 
+        // https://github.com/rust-lang/socket2/blob/3a938932829ea6ee3025d2d7a86c7b095c76e6c3/src/sockaddr.rs#L277-L287
         SocketAddr::V4(addr) => Box::into_raw(Box::new(libc::sockaddr_in {
             sin_len: std::mem::size_of::<libc::sockaddr_in>() as u8,
-            sin_family: libc::AF_INET as u8,
-            sin_port: addr.port(),
-            sin_addr: libc::in_addr {
-                s_addr: u32::from(*addr.ip()),
-            },
-            sin_zero: [0i8; 8],
+            sin_family: libc::AF_INET as libc::sa_family_t,
+            sin_port: addr.port().to_be(),
+            sin_addr: to_in_addr(addr.ip()),
+            sin_zero: Default::default(),
         })) as *mut c_void,
+        // See reference conversion from socket2:
+        // https://github.com/rust-lang/socket2/blob/3a938932829ea6ee3025d2d7a86c7b095c76e6c3/src/sockaddr.rs#L314-L331
         SocketAddr::V6(addr) => Box::into_raw(Box::new(libc::sockaddr_in6 {
             sin6_len: std::mem::size_of::<libc::sockaddr_in6>() as u8,
-            sin6_family: libc::AF_INET6 as u8,
-            sin6_port: addr.port(),
-            sin6_flowinfo: 0,
-            sin6_addr: libc::in6_addr {
-                s6_addr: addr.ip().octets(),
-            },
-            sin6_scope_id: 0,
+            sin6_family: libc::AF_INET6 as libc::sa_family_t,
+            sin6_port: addr.port().to_be(),
+            sin6_flowinfo: addr.flowinfo(),
+            sin6_addr: to_in6_addr(addr.ip()),
+            sin6_scope_id: addr.scope_id(),
         })) as *mut c_void,
     };
 
     unsafe { Box::from_raw(ptr as *mut _) }
 }
 
+// Reference implementation:
+// https://github.com/rust-lang/socket2/blob/3a938932829ea6ee3025d2d7a86c7b095c76e6c3/src/sys/unix.rs#L1356-L1363
+const fn to_in_addr(addr: &std::net::Ipv4Addr) -> libc::in_addr {
```

Comment:
It is usually considered more idiomatic to take small `Copy` types by value rather than reference. I think Clippy will suggest this is changed to `addr: Ipv4Addr`? I realize this is how `socket2` implemented it, but we don't need our version to be verbatim.

Reply:
Clippy has no complain on passing reference possibly due to function being const and maybe always inlined? Anyway I've changed code and inlined this functions manually preserving the link to reference implementation.

Reply:
Huh. The lint about this seems to be put at the pedantic level, not enabled by default, and only trigger for arguments up to 32 bit in size... Right now I also don't find it in the API design guidelines. I know this is a known best practice in general, but currently can't find any good source. Anyway, it was a nitpick only.